src/patternfly/components/Button/button.scss

```diff
@@ -110,6 +110,7 @@
 
   // Styles for an icon in button
   --pf-c-button__icon--MarginRight: var(--pf-global--spacer--xs);
+  --pf-c-button__text--icon--MarginLeft: var(--pf-global--spacer--xs);
```

Comment:
Instead of `.pf-c-button__text + . pf-c-button__icon`, what about:

```
// change during breaking change period
--pf-c-button--item--MarginRight: var(--pf-c-button__icon--MarginRight);
```